TUTORIAL.md

```diff
@@ -1950,200 +2038,69 @@ type that accumulates more and more fields as it progresses through a series of
-example = \tag ->
-    when tag is
-        Foo str -> Str.isEmpty str
-        Bar bool -> bool
-        _ -> Bool.false
+colorOrOther : [Red, Green]* -> String
+colorOrOther = \color ->
+    when color is
+        Red -> "red"
+        Green -> "green"
+        _ -> "other"
 ```
 
-```coffee
-example : [Foo Str, Bar Bool] -> Bool
-example = \tag ->
-    when tag is
-        Foo str -> Str.isEmpty str
-        Bar bool -> bool
-```
+Two things have changed compared to the first example.
+1. The `when color is` now has an extra branch: `_ -> "other"`
+2. Since this branch matches any tag, the type annotation for the `color` argument changed from the closed union `[Red, Green]` to the _open union_ `[Red, Green]*`.
 
-Similarly to how there are open records with a `*`, closed records with nothing,
-and constrained records with a named type variable, we can also have *constrained tag unions*
-with a named type variable. Here's an example:
+Also like with open records, you can name the type variable in an open tag union. For example:
 
 ```coffee
-example : [Foo Str, Bar Bool]a -> [Foo Str, Bar Bool]a
-example = \tag ->
-    when tag is
-        Foo str -> Bar (Str.isEmpty str)
-        Bar _ -> Bar Bool.false
+stopGoOther : [Red, Green]a -> [Stop, Go]a
+stopGoOther = \color ->
+    when color is
+        Red -> Stop
+        Green -> Go
         other -> other
 ```
 
-This type says that the `example` function will take either a `Foo Str` tag, or a `Bar Bool` tag,
-or possibly another tag we don't know about at compile time - and it also says that the function's
-return type is the same as the type of its argument.
+You can read this type annotation as "`stopGoOther` takes either a `Red` tag, a `Green` tag, or some other tag. It returns either a `Stop` tag, a `Go` tag, or any one of the tags it received in its argument."
 
-So if we give this function a `[Foo Str, Bar Bool, Baz (List Str)]` argument, then it will be guaranteed
-to return a `[Foo Str, Bar Bool, Baz (List Str)]` value. This is more constrained than a function that
-returned `[Foo Str, Bar Bool]*` because that would say it could return *any* other tag (in addition to
-the `Foo Str` and `Bar Bool` we already know about).
+So let's say you called this `stopGoOther` function passing `Foo "hello"`. Then the `a` type variable would be the closed union `[Foo Str]`, and `stopGoOther` would return a union with the type `[Stop, Go][Foo Str]` - which is equivalent to `[Stop, Go, Foo Str]`.
 
-If we removed the type annotation from `example` above, Roc's compiler would infer the same type anyway.
-This may be surprising if you look closely at the body of the function, because:
+> **Aside:** You can annotate a function as returning an open union with no named type variable -
+> such as `Str -> [Red, Green]*` - but since returning doesn't involve pattern matching, this
+> is no different from returning a closed tag union. It's a more verbose way to say the same thing.
```

Comment:
This is not totally correct, because there is no such thing as returning a closed tag union anymore. You can only return a tag union whose "open-ness" is inferred.

I would suggest removing this entirely, in part because the compiler will now issue a warning if you have an annotation like `Str -> [Red, Green]*` saying that the `*` is unnecessary. Or, perhaps rephrasing to something like

> Functions are always inferred to return open unions, since the caller of a function can use a return value in any context where a larger open union might be expected.

Reply:
I forgot about the warning! I'll just delete it then. 👍